orderbook/src/orderbook.rs

```diff
@@ -247,25 +263,16 @@ fn solvable_orders(mut orders: Vec<Order>, balances: &HashMap<(H160, H160), U256
     result
 }
 
-async fn new_order_has_balance(order: &Order, balance_fetcher: &dyn BalanceFetching) -> bool {
-    balance_fetcher
-        .register(order.order_meta_data.owner, order.order_creation.sell_token)
-        .await;
-    let owner = order.order_meta_data.owner;
-    let token = order.order_creation.sell_token;
-    balance_fetcher.register(owner, token).await;
-    let minimum_balance = if order.order_creation.partially_fillable {
-        1.into()
+// Mininum balance user must have in sell token for order to be accepted. None if no balance is
+// sufficient.
+fn minimum_balance(order: &Order) -> Option<U256> {
+    if order.order_creation.partially_fillable {
+        Some(U256::from(1))
```

Comment:
Doesn't this potentially lead to problems if the user can transfer N tokens, but has a balance of N + M, where the balance check would pass (since the transfer check would succeed) but the solver would be given an incorrect available balance of N + M?

Reply:
But then you can't place partially fillable orders for more than your current balance. Agree it's not perfect but not sure which the better tradeoff is. Also depends on the decision what amount we are going to check for our regular balance updates (see first thread).

Reply:
Yeah, its a tough one. I'm inclined to just require the entire remaining amount for now as its "safer" (doesn't produce invalid settlements). On other hand, I do agree that in the long term, the requirement should be relaxed.

Since we don't do partially fillable orders in the FE now anyways, I don't think its super important.

Reply:
From my point of view this doesn't block merging the PR (since the FE doesn't really support these kinds of orders anyway for the moment). Just something to be aware of.